src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
         }
     }
 
+    /**
+     * Find any eligible jobs that qualify for an automated job retry and schedule their retry
+     *
+     * @throws ValidationException
+     */
+    protected function retryEligibleJobs(int $queueType): void
+    {
+        // Find any broken jobs that qualify for an automated job retry
+        $jobRetrySentinel = (int) static::config()->get('job_retry_sentinel');
+        $jobRetryLimit = (int) static::config()->get('job_retry_limit');
+        $jobRetryStatus = (array) static::config()->get('job_retry_status');
+
+        $jobStatusMap = [];
+        $jobRetrySentinel = max(0, $jobRetrySentinel);
+
+        // Job retries feature is disabled
+        if ($jobRetryLimit <= 0) {
+            return;
+        }
+
+        $validJobStatuses = QueuedJobDescriptor::singleton()->getJobStatusValues();
+
+        foreach ($jobRetryStatus as $retryStatus => $targetStatus) {
+            // Skip any disabled status conditions
+            if (!$targetStatus) {
+                continue;
+            }
+
+            // Skip any invalid status conditions
+            if (!in_array($targetStatus, $validJobStatuses)) {
+                continue;
+            }
+
+            $jobStatusMap[$retryStatus] = $targetStatus;
+        }
+
+        $jobStatusConditions = array_keys($jobStatusMap);
```

Comment:
```suggestion
        $retryJobStatuses = array_keys($jobStatusMap);
```
or `failedJobStatuses` or `stuck_job_statuses` (using the terminology in the docs) or something else that makes it clear "these are the statuses for jobs that we want to retry"